### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by aminger-jan**

### CODE REVIEW

File: lib/ui/screens/profile/profile_view.dart [hunk 1]
```diff
@@ -1,12 +1,125 @@
...
+  static const double _containerSize = 35;
+  static const double _positionAvatarIcon = 10;
+
+  const ProfileView({
+    super.key,
+  });
 
   @override
   Widget build(BuildContext context) => Scaffold(
         appBar: AppBar(
```
Comment (SamuelBehrmann) [inline lib/ui/screens/profile/profile_view.dart:24]: **suggestion:** Use `CustomAppBar`

File: lib/ui/screens/profile/profile_view.dart [hunk 2]
```diff
@@ -1,12 +1,125 @@
...
+                        labelText: 'Password',
+                        hintText: 'Edit your password',
+                      ),
+                      SizedBox(height: _emptySpaceAfterPassword),
+                    ],
+                  ),
+                ],
+              ),
+            ),
+            CustomButton(text: 'Logout', onPressed: () {}),
```
Comment (SamuelBehrmann) [inline lib/ui/screens/profile/profile_view.dart:62]: **suggestion:** Call controller method

File: lib/ui/screens/profile/profile_view.dart [hunk 3]
```diff
@@ -1,12 +1,125 @@
...
+                    color: Theme.of(context).colorScheme.outline,
+                    shape: BoxShape.circle,
+                  ),
+                  child: IconButton(
+                    icon: Icon(
+                      Icons.create_outlined,
+                      color: Theme.of(context).colorScheme.onPrimary,
+                    ),
+                    onPressed: () {
+                      // TODO: implementation
```
Comment (SamuelBehrmann) [inline lib/ui/screens/profile/profile_view.dart:95]: **suggestion:** Call controller method

File: lib/ui/screens/profile/profile_view.dart [hunk 4]
```diff
@@ -1,12 +1,125 @@
...
+            ],
+          ),
+        ),
+      );
+
+  Widget _buildNameAndDescription() => Builder(
+        builder: (BuildContext context) => Column(
+          children: <Widget>[
+            Text(
+              dummyUser.name,
```
Comment (SamuelBehrmann) [inline lib/ui/screens/profile/profile_view.dart:110]: **suggestion:** Use actual data from the model

File: lib/ui/screens/profile/services/profile_navigation_service.dart [hunk 5]
```diff
@@ -0,0 +1,4 @@
+abstract interface class ProfileNavigationService {
```
Comment (SamuelBehrmann) [inline lib/ui/screens/profile/services/profile_navigation_service.dart:1]: **question:** What does the navigation service have to do with logout and editprofile ?

File: lib/ui/widgets/custom_button.dart [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/material.dart';
+
+class CustomButton extends StatelessWidget {
```
Comment (SamuelBehrmann) [inline lib/ui/widgets/custom_button.dart:3]: **suggestion:** Move the Styling to the theme then this component becomes obsolete

File: lib/ui/widgets/custom_text_form_field.dart [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    this.icon,
+  });
+
+  @override
+  CustomTextFormFieldState createState() => CustomTextFormFieldState();
+}
+
+class CustomTextFormFieldState extends State<CustomTextFormField> {
+  static const EdgeInsets _contentPadding = EdgeInsets.all(15);
+  static const EdgeInsets _paddingBetweenLabelField = EdgeInsets.only(top: 10);
```
Comment (SamuelBehrmann) [inline lib/ui/widgets/custom_text_form_field.dart:23]: **suggestion:** We should always use multiple of 4 for layout and multiple of 2 for sizes. 
-> padding of 16

File: lib/ui/widgets/custom_text_form_field.dart [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = TextEditingController();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
```
Comment (SamuelBehrmann) [inline lib/ui/widgets/custom_text_form_field.dart:34]: **praise:** Very nice !

File: lib/ui/widgets/custom_text_form_field.dart [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+          ),
+          maxLines: 1,
+        ),
+      ),
+    );
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: widgets,
+    );
```
Comment (SamuelBehrmann) [inline lib/ui/widgets/custom_text_form_field.dart:70]: **praise:** Nice idea, but lets try to not use mutable components there is a better way e.g.: 
```
 @override
  Widget build(BuildContext context) => Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: <Widget>[
          if (widget.includeTextField) ...<Widget>[
            Text(
              widget.labelText,
              style: Theme.of(context).textTheme.titleMedium,
            ),
            const SizedBox(height: 10),
            TextFormField(
              controller: _controller,
              decoration: InputDecoration(
                contentPadding: _contentPadding,
                hintText: widget.hintText,
                border: const OutlineInputBorder(),
                prefixIcon: widget.icon != null ? Icon(widget.icon) : null,
              ),
              maxLines: 1,
            ),
          ],
        ],
      );
```

File: lib/ui/widgets/custom_text_form_field.dart [hunk 10]
```diff
@@ -0,0 +1,72 @@
+import 'package:flutter/material.dart';
+
+class CustomTextFormField extends StatefulWidget {
+  final bool includeTextField;
+  final String labelText;
+  final String hintText;
```
Comment (SamuelBehrmann) [inline lib/ui/widgets/custom_text_form_field.dart:6]: **nitpick:** Labels and texts are always texts ;) i think we should remove the suffix

File: lib/ui/widgets/profile_password.dart [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    required this.labelText,
+    this.hintText,
+    this.includeTextField = true,
+  });
+
+  @override
+  PasswordFieldState createState() => PasswordFieldState();
+}
+
+class PasswordFieldState extends State<PasswordField> {
```
Comment (SamuelBehrmann) [inline lib/ui/widgets/profile_password.dart:19]: **suggestion:** I think we should merge this logic into the custom one. They share 99% of tf code and giving the developer the option to make any texfield obscured is a nice add on :)

File: assets/images/dummy_profile_pic.png
Comment (SamuelBehrmann) [inline assets/images/dummy_profile_pic.png:1]: **suggestion:** 
Lets make thwe eprofile image dynamic by using an url. 
For this we can use a `CachedNetworkImage` and 
https://unsplash.com/de/fotos/ein-rosa-ball-mit-schwarzen-ohren-und-augen-xyVIi4GN5Os

**Review by SamuelBehrmann — COMMENTED** [review]
Very nice Stuff 👌

File: lib/ui/screens/profile/profile_view.dart (hunk 1 above)
Comment (aminger-jan) [reply lib/ui/screens/profile/profile_view.dart:24]: done

File: lib/ui/screens/profile/profile_view.dart (hunk 3 above)
Comment (aminger-jan) [reply lib/ui/screens/profile/profile_view.dart:95]: done

File: lib/ui/screens/profile/profile_view.dart (hunk 4 above)
Comment (aminger-jan) [reply lib/ui/screens/profile/profile_view.dart:110]: done

File: lib/ui/widgets/custom_text_form_field.dart (hunk 7 above)
Comment (aminger-jan) [reply lib/ui/widgets/custom_text_form_field.dart:23]: done

File: lib/ui/widgets/custom_text_form_field.dart (hunk 8 above)
Comment (aminger-jan) [reply lib/ui/widgets/custom_text_form_field.dart:34]: thanks

File: lib/ui/widgets/custom_text_form_field.dart (hunk 10 above)
Comment (aminger-jan) [reply lib/ui/widgets/custom_text_form_field.dart:6]: true

File: lib/ui/widgets/custom_text_form_field.dart (hunk 9 above)
Comment (aminger-jan) [reply lib/ui/widgets/custom_text_form_field.dart:70]: done

File: lib/ui/widgets/profile_password.dart (hunk 11 above)
Comment (aminger-jan) [reply lib/ui/widgets/profile_password.dart:19]: done

File: assets/images/dummy_profile_pic.png
Comment (aminger-jan) [reply assets/images/dummy_profile_pic.png:1]: done

File: lib/ui/widgets/custom_button.dart (hunk 6 above)
Comment (aminger-jan) [reply lib/ui/widgets/custom_button.dart]: done

**Review by SamuelBehrmann — APPROVED** [review]

**Pull request merged**